iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -242,7 +242,7 @@ public override void UserEventTriggered(IChannelHandlerContext context, object @
                 {
                     ClientId = id,
                     HasUsername = true,
-                    Username = $"{this.iotHubHostName}/{id}/api-version={ClientApiVersionHelper.ApiVersionString}&DeviceClientType={Uri.EscapeDataString(this.productInfo.ToString())}",
+                    Username = $"{this.iotHubHostName}/{id}/?api-version={ClientApiVersionHelper.ApiVersionString}&DeviceClientType={Uri.EscapeDataString(this.productInfo.ToString())}",
```

Comment:
Should we remove the '/' here?

Reply:
We shouldn't. The '?' should always be in the first character of the last segment. Otherwise, we shouldn't be able to identify if '?' is part of the id or the beginning of the property bag.

Reply:
Fair enough.